GVFS/GVFS.UnitTests/Git/GitProcessTests.cs

```diff
@@ -14,5 +14,104 @@ public void TryKillRunningProcess_NeverRan()
             GitProcess process = new GitProcess(new MockGVFSEnlistment());
             process.TryKillRunningProcess().ShouldBeTrue();
         }
+
+        [TestCase]
+        public void ResultHasNoErrors()
+        {
+            GitProcess.Result result = new GitProcess.Result(
+                string.Empty,
+                string.Empty,
+                0);
+
+            result.ExitCodeIsFailure.ShouldBeFalse();
+            result.StderrContainsErrors().ShouldBeFalse();
+        }
+
+        [TestCase]
+        public void ResultHasWarnings()
+        {
+            GitProcess.Result result = new GitProcess.Result(
+                string.Empty,
+                "Warning: this is fine.\n",
+                0);
+
+            result.ExitCodeIsFailure.ShouldBeFalse();
+            result.StderrContainsErrors().ShouldBeFalse();
+        }
+
+        [TestCase]
+        public void ResultHasNonWarningErrors_SingleLine_AllWarnings()
+        {
+            GitProcess.Result result = new GitProcess.Result(
+                string.Empty,
+                "warning: this line should not be considered an error",
+                1);
+
+            result.ExitCodeIsFailure.ShouldBeTrue();
+            result.StderrContainsErrors().ShouldBeFalse();
+        }
+
+        [TestCase]
+        public void ResultHasNonWarningErrors_Multiline_AllWarnings()
+        {
+            GitProcess.Result result = new GitProcess.Result(
+                string.Empty,
+                @"warning: this line should not be considered an error
+WARNING: neither should this.",

```

Comment:
Related to Saeed's comment, add a test with blank lines between warning lines?

Reply:
Yep, thanks